.github/workflows/pr-pull.yaml

```diff
@@ -0,0 +1,94 @@
+name: Pull Request
+run-name: ${{ github.event.pull_request.title }} pull request run 🚀
+on:
+  workflow_dispatch:
+  pull_request:
+    paths-ignore:
+      - '**.md'
+    branches: [master]
+    types: [ opened, synchronize, reopened, labeled, unlabeled ]
+permissions:
+  contents: read
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+jobs:
+  check-format:
+    name: Check Formatting 🔍
+    uses: ./.github/workflows/check-format.yaml
+    permissions:
+      contents: read
+
+  build-project:
+    name: Build Project 🧱
+    uses: ./.github/workflows/build-project.yaml
+    secrets: inherit
+    permissions:
+      contents: read
+
+  compatibility-validation:
+    name: Validate Compatibility Data 🕵️
+    if: github.base_ref == 'master'
+    runs-on: ubuntu-22.04
+    permissions:
+      checks: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check for Changed Files ✅
+        uses: ./.github/actions/check-changes
+        id: checks
+        with:
+          baseRef: origin/${{ github.base_ref }}
+          checkGlob: plugins/win-capture/data/*.json
+
+      - name: Check for Invalid Compatibility Data 📉
+        if: fromJSON(steps.checks.outputs.passed)
+        uses: ./.github/actions/compatibility-validator
+
+  services-validation:
+    name: Validate Services Data 🕵️
+    if: github.base_ref == 'master'
+    runs-on: ubuntu-22.04
+    permissions:
+      checks: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check for Changed Files ✅
+        uses: ./.github/actions/check-changes
+        id: checks
+        with:
+          baseRef: origin/${{ github.base_ref }}
+          checkGlob: plugins/rtmp-services/data/*.json
+
+      - name: Check for defunct services 📉
```

Comment:
Correct, defunct services are only checked in scheduled workflow runs or manual dispatches. That's the current setup as well.

Reply:
I more meant that the name "Check for defunct services" is misleading in this invocation, because the invoked call does not, in fact, check for defunct services. Am I mistaken? Do the current invocations also have this issue with a name/label?

Reply:
Right, yeah I just copied the block around but there's no need for the name to be identical.